File: scripts/remove-bg.py
from PIL import Image

input_path = "/vercel/share/v0-project/public/images/ph-outline-map.png"
output_path = "/vercel/share/v0-project/public/images/ph-map-transparent.png"

# Open image (PIL auto-detects format regardless of extension)
img = Image.open(input_path).convert("RGBA")
pixels = img.load()

threshold = 220

for y in range(img.height):
    for x in range(img.width):
        r, g, b, a = pixels[x, y]
        # If pixel is near-white, make fully transparent
        if r > threshold and g > threshold and b > threshold:
            pixels[x, y] = (r, g, b, 0)

img.save(output_path, "PNG")
print(f"Done! Transparent PNG saved to {output_path}")
print(f"Size: {img.width}x{img.height}")

File: scripts/remove-bg.js
import sharp from "sharp";
import path from "path";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const inputPath = path.join(__dirname, "../public/images/ph-outline-map.png");
const outputPath = path.join(__dirname, "../public/images/ph-outline-map-transparent.png");

async function removeBg() {
  const image = sharp(inputPath);
  const metadata = await image.metadata();
  const { width, height } = metadata;

  // Get raw pixel data with alpha channel
  const raw = await image.ensureAlpha().raw().toBuffer();

  const pixels = Buffer.from(raw);
  const threshold = 200; // Pixels whiter than this become transparent

  for (let i = 0; i < pixels.length; i += 4) {
    const r = pixels[i];
    const g = pixels[i + 1];
    const b = pixels[i + 2];

    // If pixel is near-white, make it fully transparent
    if (r > threshold && g > threshold && b > threshold) {
      pixels[i + 3] = 0; // Set alpha to 0
    }
  }

  await sharp(pixels, {
    raw: { width, height, channels: 4 },
  })
    .png()
    .toFile(outputPath);

  console.log("Done! Transparent PNG saved to", outputPath);
  console.log("Image dimensions:", width, "x", height);
}

removeBg().catch(console.error);
